### PR TITLE
Reduce calls to set the system bars' style when scrolling vertically in the HomeFragment

### PR DESCRIPTION
**Pull request by min7-i**
Currently, when scrolling vertically in the HomeFragment `checkToSetSystemBarStyle` is called whenever the vertical location (translation y) of the top app bar changes. As a result, multiple unnecessary calls are made to [update the system bars' style](https://github.com/breezy-weather/breezy-weather/blob/22da48c72f2421e96300a160af059945ce1037bf/app/src/main/java/org/breezyweather/ui/main/MainActivity.kt#L762).

The PR simplifies the previous solution and only calls `checkToSetSystemBarStyle` when the top app bar is actually starting to collapse or is fully expanded. This still ensures that the status bar is shaded when the top app bar is collapsing/collapsed and that it stays transparent when expanded.

Successfully tested on the following devices: Google Pixel 6a (Android 15) and LG G6 (Android 9).

### CODE REVIEW

File: app/src/main/java/org/breezyweather/ui/main/fragments/HomeFragment.kt [hunk 1]
```diff
@@ -551,11 +543,8 @@ class HomeFragment : MainModuleFragment() {
...
             // set system bar style.
-            mTopChanged = if (mFirstCardMarginTop <= 0) {
-                (binding.appBar.translationY != 0f) != (mLastAppBarTranslationY != 0f)
-            } else {
-                true
-            }
+            // Only set mTopChanged to true when the app bar actually starts collapsing or stops expanding.
+            mTopChanged = (binding.appBar.translationY == 0f) != (mLastAppBarTranslationY == 0f)
             topOverlap = binding.appBar.translationY != 0f
             if (mTopChanged!!) {
```
Comment (papjul) [inline app/src/main/java/org/breezyweather/ui/main/fragments/HomeFragment.kt:549]: Can we inline it directly here? Looks like the `mTopChanged` variable isn't used anywhere else (besides an useless reset).

Other than that, tried a few weird moves to test edge cases, and found nothing wrong.

Comment (min7-i) [reply app/src/main/java/org/breezyweather/ui/main/fragments/HomeFragment.kt]: You're right. `onScrolled` is called in `postReset` anyway, so the system bars are still updated correctly when `postReset` is called.

I updated the PR accordingly.

**Comment by papjul** [comment]
Thank you!
